modules/gapi/test/common/gapi_imgproc_tests_common.hpp

```diff
@@ -14,6 +14,105 @@
 
 namespace opencv_test
 {
+// Draw random ellipses on given mat of given size and type
+static void initMatForFindingContours(cv::Mat& mat, const cv::Size& sz, const int type)
+{
+    cv::RNG& rng = theRNG();
+    mat = cv::Mat(sz, type, cv::Scalar::all(0));
+    size_t numEllipses = rng.uniform(1, 10);
+
+    for( size_t i = 0; i < numEllipses; i++ )
+    {
+        cv::Point center;
+        cv::Size  axes;
+        center.x     = rng.uniform(0, sz.width);
+        center.y     = rng.uniform(0, sz.height);
+        axes.width   = rng.uniform(2, sz.width);
+        axes.height  = rng.uniform(2, sz.height);
+        int color    = rng.uniform(1, 256);
+        double angle = rng.uniform(0., 180.);
+        cv::ellipse(mat, center, axes, angle, 0., 360., color, 1, FILLED);
+    }
+}
+
+enum WithHierarchy {NO_HIERARCHY, WITH_HIERARCHY};
```

Comment:
WithHierarchy can be NO_HIERARCHY. It is strange.
```
enum OptionalOutput {
     NONE,
     HIERARCHY
};
```

M.B.

Reply:
Can agree with you. Naming is not my strength :)
One of the ideas though was to make the name unique not to accidentally conflict with some possible `cv::OptionalOutput` already existing, or, what is worse, to cause some failures in the future when such the name would be accidentally added.. I hope what I mean is clear
So, `OptionalFindContoursOutput` is good enough I hope